### PR TITLE
Correct besseli(nu, 0) & besselix(nu, 0) with nu<0 and nu an Int.

### PR DESCRIPTION
**Pull request by matthew-wingate**
An AmosException is thrown for besseli(nu, 0) or besselix(nu, 0)
due to the call to _besselk.
However, for integer nu, besseli(nu, 0) = besseli(-nu,0) = 0 and
similar for besselix.  A test is added before calling besselk.

### CODE REVIEW

**Comment by matthew-wingate** [comment]
I see these tests have failed because runtests.jl needs updating to account for correcting the nu=-1, z=0 behavior of besselix.  I also believe besseljx(-1,0) should be equal to zero, not divergent.  When I get a bit more time I will go through carefully and systematically to check all the bessel functions for these arguments.

**Comment by matthew-wingate** [comment]
It looks like the coverage test is failing due to code I didn't change.  Please advise.

**Comment by ararslan** [comment]
The drop in coverage could mean that some of the branches added here are keeping some code paths from getting hit but who knows; coverage is a fickle beast. I'll retrigger CI to see if it changes.

**Comment by matthew-wingate** [comment]
I've tried my best.  The coverage of src/bessel.jl is 97.02%.  It would be great if my bug-fix and improvements of src/bessel.jl and tests/runtests.jl could be merged, fixing issue 33/35.  It would be a shame if they were held back due to coverage failures by code already in the master repo.

According to the coverage report, the problem seems to be associated with the macro `@horner`.  I tried a couple (probably naive) edits to src/sincosint.jl.  These changes can be discarded.  I'm out of ideas and hope someone else can improve the coverage there.

**Comment by ararslan** [comment]
Thanks for the contribution!

File: src/sincosint.jl [hunk 1]
```diff
@@ -18,7 +18,7 @@ using Base.Math.@horner
 function sinint(x::Float64)
     t = x*x
     if t ≤ 36.0
-        return x * @horner(t, 1.00000000000000000000E0,
```
Comment (musm) [inline src/sincosint.jl]: why change these, there was nothing wrong with just returning the result directly

Comment (matthew-wingate) [inline src/sincosint.jl]: I was just trying to understand why the coverage tests flagged these lines as being untested. In the end it had no effect so I agree there is nothing gained by the change. I just forgot to undo the change in my branch, sorry.

Comment (musm) [inline src/sincosint.jl]: ok thanks for letting me know, I opened a PR to go back. Coverage information is typically a little noisy and not always exact.
